### PR TITLE
Fix banishing magic loss and actions

### PR DESCRIPTION
**Pull request by jdevnull**
(1) Temp magic loss is recovered 1 per mud hour.
(2) Cap temp magic loss.
(3) Spirits don't otherwise act when in banishment clashes.

Discussed at https://discord.com/channels/564618629467996170/1076958471288787105

- Khai

### CODE REVIEW

File: src/fight.cpp [hunk 1]
```diff
@@ -5946,7 +5946,7 @@ void perform_violence(void)
       if (success < 0) {
         GET_EXTRA(mage) = 0;
         GET_EXTRA(spirit) = 1;
-        GET_TEMP_MAGIC_LOSS(mage) -= success;
+        GET_TEMP_MAGIC_LOSS(mage) += MAX(-success, GET_MAG(mage));
```
Comment (luciensadi) [inline src/fight.cpp:5949]: Important to note here is that GET_MAG is 100x the value you'd expect it to be since it's an integer that masquerades as a floating-point number. GET_MAG 600 is equivalent to magic 6.00.

I think the logic here isn't quite right regardless though-- in the case of `success = -1, GET_MAG(mage)/100 = 6`, this would add 6 to their temp magic loss var every time without actually capping the result.

Maybe this: `GET_TEMP_MAGIC_LOSS(mage) = MAX(GET_TEMP_MAGIC_LOSS(mage) - success, GET_MAG(mage) / 100);`

**Review by luciensadi — CHANGES_REQUESTED** [review]

File: src/fight.cpp (hunk 1 above)
Comment (jdevnull) [reply src/fight.cpp]: Oops, when I switched the signs, I forgot to switch MAX to MIN, which I think is what we want.

Rubber ducking - if successes is negative, then magic loss should increase by a positive number, where that positive number should not exceed that character's current magic rating.

Comment (luciensadi) [reply src/fight.cpp]: Note that you're doing a `+=` on the result of your max/min, so whatever value you choose will be added to the current temp loss instead of overwriting it. That's why the version I suggested is a flat assignment `=`.

Comment (jdevnull) [reply src/fight.cpp]: Yeah, the problem is affect_total does:
`  GET_MAG(ch) -= MIN(GET_MAG(ch), GET_TEMP_MAGIC_LOSS(ch) * 100);
`
So doing a flat assignment comparing to GET_MAG won't work. Comparing against GET_REAL_MAG will work, though I think that's functionally identical to the current solution when taking affect_total into account.

Comment (jdevnull) [reply src/fight.cpp]: Oh wait, that line in affect_total also is wrong.

Comment (jdevnull) [reply src/fight.cpp]: Nope, nevermind, missed the part where affect_totals set `ch->aff_abils = ch->real_abils;`

Comment (jdevnull) [reply src/fight.cpp]: My current `GET_TEMP_MAGIC_LOSS(mage) += MIN(-success, GET_MAG(mage) / 100);` will let GET_MAG hit zero, but when a power focus is in play, that means GET_TEMP_MAGIC_LOSS can become greater than real magic.

Alternate solution `GET_TEMP_MAGIC_LOSS(mage) = MIN(GET_TEMP_MAGIC_LOSS(mage) - success, GET_REAL_MAG(mage) / 100);` will more correctly cap temp magic loss to real magic, but that would also mean using a power focus prevents a character from hitting zero current mag against a tougher spirit and so prevents the spirit from knocking out the mage and ending the banishing attempt.

Comment (jdevnull) [reply src/fight.cpp]: Alright, new solution follows https://discord.com/channels/564618629467996170/564621949842751508/1077145056357666826, capping temp magic loss = real magic and knocking out the character at that point, regardless of power focus.

**Review by luciensadi — APPROVED** [review]
Looks good, thanks!
